sentry/src/routes/campaign.rs

```diff
@@ -17,28 +38,320 @@ pub async fn create_campaign<A: Adapter>(
+
+    let accounting_spent = get_accounting_spent(app.pool.clone(), &campaign.creator, &campaign.channel.id()).await?;
+
+    // TODO: AIP#61: Update when changes to Spendable are ready
+    let latest_spendable = fetch_spendable(app.pool.clone(), &campaign.creator, &campaign.channel.id()).await?;
+    let remaining_for_channel = get_total_remaining_for_channel(&accounting_spent, &latest_spendable)?;
+
+    if campaign.budget > remaining_for_channel {
+        return Err(ResponseError::Conflict("Not Enough budget for campaign".to_string()));
+    }
+
+    // If the channel is being created, the amount spent is 0, therefore remaining = budget
+    update_remaining_for_campaign(&app.redis.clone(), campaign.id, campaign.budget).await?;
 
     // insert Campaign
-    
-    // match insert_campaign(&app.pool, &campaign).await {
-    //     Err(error) => {
-    //         error!(&app.logger, "{}", &error; "module" => "create_channel");
-
-    //         match error {
-    //             PoolError::Backend(error) if error.code() == Some(&SqlState::UNIQUE_VIOLATION) => {
-    //                 Err(ResponseError::Conflict(
-    //                     "channel already exists".to_string(),
-    //                 ))
-    //             }
-    //             _ => Err(error_response),
-    //         }
-    //     }
-    //     Ok(false) => Err(error_response),
-    //     _ => Ok(()),
-    // }?;
-
-    let create_response = SuccessResponse { success: true };
+    match insert_campaign(&app.pool, &campaign).await {
+        Err(error) => {
+            error!(&app.logger, "{}", &error; "module" => "create_campaign");
+            return Err(ResponseError::Conflict("campaign already exists".to_string()));
+        }
+        Ok(false) => Err(error_response),
+        _ => Ok(()),
+    }?;
 
     Ok(success_response(serde_json::to_string(&campaign)?))
 }
+
+pub async fn update_campaign<A: Adapter>(
+    req: Request<Body>,
+    app: &Application<A>,
+) -> Result<Response<Body>, ResponseError> {
+    let body = hyper::body::to_bytes(req.into_body()).await?;
+
+    let campaign = serde_json::from_slice::<Campaign>(&body)
+        .map_err(|e| ResponseError::FailedValidation(e.to_string()))?;
+
+    let error_response = ResponseError::BadRequest("err occurred; please try again later".to_string());
+
+    // modify Campaign
+
+    match modify_campaign(&app.pool, &campaign, &app.redis).await {
+        Err(error) => {
+            error!(&app.logger, "{:?}", &error; "module" => "update_campaign");
+            return Err(ResponseError::Conflict("Error modifying campaign".to_string()));
+        }
+        Ok(false) => Err(error_response),
+        _ => Ok(()),
+    }?;
+
+    Ok(success_response(serde_json::to_string(&campaign)?))
+}
+
+// TODO: Double check redis calls
+async fn get_spent_for_campaign(redis: &MultiplexedConnection, id: CampaignId) -> Result<UnifiedNum, ResponseError> {
```

Comment:
We should not use a `spent` key. The Spent can be calculated by `campaign.budget - campaignRemaining` if we need it.

- [ ] Remove function